### PR TITLE
s3: Add progress tracker support to s3 downloads

### PR DESCRIPTION
**Pull request by mattrobenolt**
This was oddly missing entirely and I was real sad. It seems to only be
supported on HTTP right now.

### CODE REVIEW

File: get_s3.go [hunk 1]
```diff
@@ -173,7 +173,15 @@ func (g *S3Getter) getObject(ctx context.Context, client *s3.S3, dst, bucket, ke
 		return err
 	}
 
-	return copyReader(dst, resp.Body, 0666, g.client.umask())
+	body := resp.Body
+
+	if g.client.ProgressListener != nil {
```
Comment (azr) [inline get_s3.go:178]: Watch out, the tests paniced here for me:
```suggestion
	if g.client != nil && g.client.ProgressListener != nil {
```

**Review by azr — APPROVED** [review]
LGTM after applying my suggestion, thanks

**Comment by mattrobenolt** [comment]
Interesting, I wonder if that case _only_ applies during test suite, since I couldn't find any path where `client` could be `nil` in normal use. Lemme add that check tho.

File: get_s3.go (hunk 1 above)
Comment (mattrobenolt) [reply get_s3.go]: Added.

**Comment by azr** [comment]
Thanks ! Yes, the tests test things in isolation 🙂 .

**Pull request merged**